README.md

```diff
@@ -209,10 +209,57 @@ Or
 Branch branch = Branch.getInstance(getApplicationContext(), "your test branch key"); // replace with your actual branch key
 ```
 
-Either way, we recommend you put a `//TODO` to remind you to change back to live app during deployment later. 
+Either way, we recommend you put a `//TODO` to remind you to change back to live app during deployment later.
 Also, note the Branch object is singleton, so you can and should still use `Branch.getInstance(getApplicationContext())` in all the other places (see examples below).
 
+#### Automatic Session Management
+Starting from 1.5.7, there is no need for initialising and closing session with the new Automatic Session Management. Automatic Session Management can work only with API level 14 and above.
+So make sure that your `minSdkVersion` is 14 or above.
+
+```xml
+<uses-sdk
+	android:minSdkVersion="14"
+	   ------------        />
+```
+
+Branch SDK can do session management for you if you do one of the following.
+
+1) If you are not creating an Application class, declare `BranchApp` as your application class in your manifest.
+
+```xml
+ <application
+-----
+android:name="io.branch.referal.BranchApp">
+```
+
+2) If you already have an Application class then extend your application class with `BranchApp`.
```

Comment:
This is not really 1 and 2 or 3 for steps. These should be ORs - let me pull down and modify layout.